DataFormats/Provenance/src/IndexIntoFile.cc

```diff
@@ -969,7 +1009,7 @@ namespace edm {
         }
       }
     } else if (type_ == kLumi) {
-      if (indexToLumi_ + 1 == size_) {
+      if (indexToLumi_ + 1 == indexedSize()) {
```

Comment:
I could delete the function size() and only use indexedSize()
(and possibly rename indexedSize -> size). I am not feeling
particularly strong about this.

I can think of two reasons not to do that.

First, size() is a non-virtual function and so maybe slightly faster than
the virtual function indexedSize(). In the old iterator classes named
IndexIntoFileItrNoSort and IndexIntoFileItrSorted, the functions size()
and indexedSize() do exactly the same thing. The performance improvement
related to avoiding this virtual function may be negligible though (or maybe
the compiler even optimizes it away). I didn't profile.

If I try to minimize the changes to the existing iterator classes
(hoping to avoid new bugs), then I could keep the size() function
exactly as is. It is used in the older iterator classes and also some
tests. I think that was the original motivation. Just an attempt to
avoid disturbing the existing code.

What do you think?

Reply:
Let's go with what you have.